### DISCUSSION TITLE
Assigning variables in on_[macosversion] blocks

### DISCUSSION THREAD

**Discussion by unitof** [Homebrew/discussions#5431] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

The audio application developer Rouge Amoeba has been gradually updating their applications ([Piezo](https://weblog.rogueamoeba.com/2024/03/13/piezo-ark-update-launches/), [Audio Hijack](https://weblog.rogueamoeba.com/2024/04/05/our-new-installer-free-setup-comes-to-audio-hijack/), [Loopback](https://weblog.rogueamoeba.com/2024/06/07/now-you-can-install-loopback-in-under-one-minute/)) to support macOS's new Audio Routing Kit framework for less invasive installation. ARK is available on macOS 14.5 and later. For macOS 14.4.x and below, Rogue Amoeba is continuing development a separate version track of each app using their legacy ACE audio engine, typically one minor version below the ARK version but with identical functionality.

The Casks now have a number of largely repeated stanzas and strings, especially the `url` and `livecheck`, and was hoping to do something like the pattern we often do with [`livecheck_folder` and `on_arch_conditional`](https://github.com/Homebrew/homebrew-cask/blob/b008d828253f964e764d3d0f4a821698da6f5563/Casks/n/notion.rb#L3), but I don't see any conditionals similar to `on_arch_conditional` for macOS versions.

Is the below permissible? Is there a nicer way to do it? It passes `audit` and `style`, but no other casks define variables as empty strings up top, so I assume it is discouraged and ugly.

```ruby
cask "loopback" do
  sha256 :no_check

  macosversion_no_dots = ""
  audio_engine = ""

  on_ventura :or_older do
    # macOS 11.0 is oldest supported version
    macosversion_no_dots = "110"
    audio_engine = "ace"

    version "2.3.3"
  end
  on_sonoma :or_newer do
    macosversion_no_dots = "145"
    audio_engine = "ark"

    version "2.4.0"
  end

  url "https://rogueamoeba.com/loopback/download-#{audio_engine}.php"
  name "Loopback"
  desc "Cable-free audio router"
  homepage "https://rogueamoeba.com/loopback/"

  livecheck do
    url "https://rogueamoeba.net/ping/versionCheck.cgi?format=sparkle&system=#{macosversion_no_dots}&bundleid=com.rogueamoeba.Loopback&platform=osx&version=#{version.no_dots}8000"
    strategy :sparkle
  end

  # code omitted

end
```

For now, I have defined `url` and `livecheck` for each `on_[macosversion]` block, in full: https://github.com/Homebrew/homebrew-cask/pull/176100/files

**Comment by SMillerDev** [comment, marked as answer]
I would put the newest version by default and override in only one block
